src/main/java/jenkins/branch/MultiBranchProjectViewHolder.java

```diff
@@ -223,7 +223,7 @@ public ViewImpl(ViewGroup owner, @NonNull SCMHeadCategory category) {
          */
         @Override
         public String getDisplayName() {
-            return category.getDisplayName().toString();
+            return category.getDisplayName().toString() + " (" + getItems().size() + ")";
```

Comment:
Should not be, as all items are in memory. There is allocation of an array and deduplication, so not the kind of thing you'd want to do in a tight loop, but in a UI request it should be reasonably cheap